synapse/rest/client/receipts.py

```diff
@@ -52,7 +54,14 @@ async def on_POST(
         if receipt_type != "m.read":
             raise SynapseError(400, "Receipt type must be 'm.read'")
 
-        body = parse_json_object_from_request(request, allow_empty_body=True)
+        user_agent = get_request_user_agent(request)
+        pattern = re.compile(r"(?:Element|SchildiChat)/1\.[012]\.")
```

Comment:
We want to compile this only once (e.g. make it a constant outside of the function) or alternatively use `re.match` without pre-compiling so that it uses the re compiled pattern cache.